src/NServiceBus.Transport.RabbitMQ.TransportTests/When_restarting_endpoint.cs

```diff
@@ -0,0 +1,108 @@
+﻿namespace NServiceBus.TransportTests
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using NUnit.Framework;
+    using Transport;
+
+    public class When_restarting_endpoint : NServiceBusTransportTest
+    {
+        [TestCase(TransportTransactionMode.None)]
+        [TestCase(TransportTransactionMode.ReceiveOnly)]
+        [TestCase(TransportTransactionMode.SendsAtomicWithReceive)]
+        [TestCase(TransportTransactionMode.TransactionScope)]
+        public async Task Should_handle_multiple_stop_calls_to_started_receiver(TransportTransactionMode transactionMode)
+        {
+            await StartPump(
+                (context, token) => Task.CompletedTask,
+                (context, token) => Task.FromResult(ErrorHandleResult.Handled),
+                transactionMode);
+
+            await receiver.StopReceive();
+            await receiver.StopReceive();
```

Comment:
I've changed the second one to an Assert.DoesNotThrowAsync, have a look if you like this better. Imo it's a bit more straining on the eye because the Assert methods are not task returning and I find that a bit ugly (especially when considering adopting this approach into the next tests) but I also don't have a strong opinion on this though.